docs/reference/files/metadata-yaml-file.rst

```diff
@@ -0,0 +1,896 @@
+            <device name>:
+
+                # (Required) The type of device requested
+                type: gpu | nvidia.com/gpu | amd.com/gpu
+
+                # (Optional) Description of the requested device
+                description: <description>
+
+                # (Optional) Minimum number of devices required
+                countmin: <n>
+
+                # (Optional) Maximum number of devices required
+                countmax: <n>
+
+        # (Optional) Extra bindings for the charm. For example binding extra network
+        # interfaces. Key only map, value must be blank. Key represents the name
+        extra-bindings:
+            # Key only map; key represents the name of the binding
+            <binding name>:
+
+        # (Optional) A set of features that must be provided by the Juju model to ensure that the charm can be successfully deployed.
+        # See https://juju.is/docs/olm/supported-features for the full list.
+        assumes:
+            - <feature_name>
+            - any-of:
+                - <feature_name>
+                - <feature_name>
+            - all-of:
+                - <feature_name>
+                - <feature_name>
+
+        # (Optional) What kind of user is required to run the charm code.
+        # It can be one of root, sudoer or non-root.
+        # Added in Juju 3.6.0. If not specified, root is assumed.
+        charm-user: <one of root, sudoer or non-root>
+
+
+``assumes``
+-----------
+
+.. Based on https://discourse.charmhub.io/t/assumes/5450#feature-names-and-version-constraints
+
+**Status:** Optional. Recommended for Kubernetes charms.
+
+**Purpose:** The ``assumes`` key  allows charm authors to explicitly state in the metadata
+of a charm various features that a Juju model must be able to provide to ensure that the
+charm can be successfully deployed on it. When a charm comes preloaded with such
+requirements, this enables Juju to perform a pre-deployment check and to display
+user-friendly error messages if a feature requirement cannot be met by the model that
+the user is trying to deploy the charm to. If the assumes section of the charm metadata
+is omitted, Juju will make a best-effort attempt to deploy the charm, and users must
+rely on the output of ``juju status`` to figure out whether the deployment was successful.
+The ``assumes`` key is available since 2.9.23.
+
+**Structure:** The key consists of a list of features that can be given either directly
+or, depending on the complexity of the condition you want to enforce, nested under one
+or both of the boolean expressions ``any-of`` or ``all-of``, as shown below. In order for a
+charm to be deployed, all entries in the ``assumes`` block must be satisfied.
+
+.. code-block:: yaml
+
+    assumes:
+        - <feature_name>
+        - any-of:
+            - <feature_name>
+            - <feature_name>
+        - all-of:
+            - <feature_name>
+            - <feature_name>
```

Comment:
This is rendering incorrectly:

![image](https://github.com/user-attachments/assets/8c7ebe23-5b00-4bb9-b996-ebe9763197ac)

Reply:
This is caused by accidental 4-space indentation.